src/main/java/gregtech/api/recipes/ingredients/nbtmatch/NBTMatcher.java

```diff
@@ -103,6 +104,47 @@ static boolean hasKey(NBTTagCompound tag, String key, int tagType) {
         return false;
     };
 
+    /**
+     * Return true if tag has an entry where the value is equal to the condition's value.
+     * If NBTTagCompound is found, evaluates recursively.
+     */
+    public static final NBTMatcher RECURSIVE_EQUAL_TO = new NBTMatcher() {
+        @Override
+        public boolean evaluate(@Nullable NBTTagCompound tag, @Nullable NBTCondition condition)  {
+            if (condition == null || condition.tagType == null) {
```

Comment:
Should we also check if `condition.nbtKey` is equal to null here?

Reply:
‘tagType‘, `nbtKey`, and `value` can only be assigned to null via `private NBTCondition()`. If `tagType` is not null, we can confirm `nbtKey` and `value` are not null, at least with current implementation.